setup/src/Magento/Setup/Model/Installer.php

```diff
@@ -947,6 +959,10 @@ public function installDataFixtures(array $request = [])
         $this->handleDBSchemaData($setup, 'data', $request);
         $this->log->log('Enabling caches:');
         $this->updateCaches(true, $frontendCaches);
+        if ($keepCacheStatuses && !empty($disabledCaches)) {
+            $this->log->log('Disabling pre-disabled caches:');
+            $this->updateCaches(false, array_keys($disabledCaches));
+        }
```

Comment:
It seems to me really bad to enable caches and disable them again. I think it's better to calculate what caches need to be re-enabled, and enable only needed

Reply:
Just fixed it myself